dspace-server-webapp/src/main/java/org/dspace/app/rest/link/HalLinkFactory.java

```diff
@@ -43,13 +43,16 @@ public List<Link> getLinksFor(HALResource halResource, Pageable pageable) throws
         return list;
     }
 
-
     protected <T> Link buildLink(String rel, T data) {
         UriComponentsBuilder uriComponentsBuilder = uriBuilder(data);
 
         return buildLink(rel, uriComponentsBuilder.build().toUriString());
     }
 
+    protected <T> Link buildLink(T data, String rel) {
```

Comment:
I agree with @tdonohue having two methods that only differ for the parameters order is not acceptable so we should give a new name to one of the two,
The reason why we need two is actually because the existing one is buggy and lead in most case to a double encoding as reported here https://github.com/DSpace/DSpace/issues/8333

Unfortunately, the proper implementation that would rely completely on the "spring-hal-library" approach lead to the introduction of template link that is itself a good thing but right now would create trouble to angular as was previously noted by @tdonohue 

For this reason I suggest to add this alternative method as the preferred one and flag the other one as deprecated / add a javadoc referencing the reported bug.

In the context of the above bug fix we could get ride of the deprecated method but it is hard to deal with it in time for 7.3.

So the javadoc for the existing method would be
```
Please note that this method could lead to double encoding, you should use the new linkTo method instead. See also
```

the javadoc for the new method, to be renamed in getLinkTo could be
```
This is a shortcut to build a Link with the provided dummy method invocation result and the specified link relation.
```

Is this ok for you @tdonohue ? or should we just drop this new method and do what we need directly in the ExernalSourceEntryHalLinkFactory?

Reply:
Resolved, I see this was moved to `ExternalSourceEntryHalLinkFactory` for now, which is fine with me.